### PR TITLE
Add test code coverage + checksums

### PR DESCRIPTION
**Pull request by aoudiamoncef**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -201,19 +201,18 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
       - uses: actions-rs/cargo@v1
         with:
-          command: install
-          args: cargo-nextest
-      - uses: actions-rs/cargo@v1
-        with:
-          command: nextest
-          args: run --features testing --retries 10
```
Comment (AurelienFT) [inline .github/workflows/ci.yml:209]: if you change this nearly 80% of our CI will fail, some tests are too sensible for the CI and so fail often. You can see it with the tests on this PR.

Comment (AurelienFT) [reply .github/workflows/ci.yml:209]: The tools you are using takes more than the current space we have in the runners

Comment (aoudiamoncef) [reply .github/workflows/ci.yml:209]: Yep, we need more powerful runners, I'll mark this PR as blocked

**Comment by aoudiamoncef** [comment]
@damip we'll upload the coverage report at the end of testing phase. This PR will be merge when we got a better runners.

**Comment by AurelienFT** [comment]
bors check

**Comment by aoudiamoncef** [comment]
@AurelienFT I did a small change to this PR. We have to run the test coverage locally as It's very heavy. Then it'll be pushed automatically. 

I added release note files checksum which is very important for the mainet binaries

**Pull request merged**